### DISCUSSION TITLE
Use L2 Sequencer Health Flag for Chainlink Oracles

### DISCUSSION THREAD

**Discussion by CalabashSquash** [tracer-protocol/perpetual-pools-contracts#284] [Ideas]
Main relevant documentation: https://docs.chain.link/docs/l2-sequencer-flag/

> It`s recommended to use L2 Sequencer Health Flag for all of the contracts on Arbitrum network!
\- Chainlink

This requires adding a `FlagsInterface` check and
```
        bool isRaised = chainlinkFlags.getFlag(FLAG_ARBITRUM_SEQ_OFFLINE);
        if (isRaised) {
                // If flag is raised we shouldn't perform any critical operations
            revert("Chainlink feeds are not being updated");
        }
```

See the above documentation for more details.

**Comment by sporejack** [comment]
> This requires adding a `FlagsInterface`

As in, add a new interface on our end for oracles or bake it into `IOracleWrapper`?

**Reply by CalabashSquash** [reply]
No. that sentence is misleading. it should read "This requires a `FlagInterface` check".

**Comment by ghost** [comment]
@greenbookwebb question from @mynameuhh : "If the sequencer is offline, do we want to revert or do we want to keep operating under the assumption that the Chainlink oracle data is correct, even though it isn't?"
